overlay/master.cpp

```diff
@@ -331,12 +333,17 @@ struct Overlay
       overlay.set_subnet6(stringify(network6.get()));
       overlay.set_prefix6(prefix6.get());
     }
+    overlay.set_enabled(enabled);
 
     return overlay;
   }
 
   Try<Network> allocate()
   {
+    if (!enabled) {
```

Comment:
There were upgrade issues whenever we tried to skip the allocation. Therefore, by design, the overlay is allocated and the rest of the cluster checks the `flag` to act on it. It is a one-time activity so it should be fine.

Reply:
Could you please elaborate on the issue with upgrade? I can add some tests to check these cases.

The current design is not fine, we all know that. Adding another ipv6 specific flag isn't improving the situation.

Reply:
The system would be in two different internal states if you enable and then disable IPv6 as compare to just keep IPv6 disabled. In the first case, you would see IPv6 configuration in the overlay endpoint whereas in later it won't be present.  This will surely add to the confusion and may lead to certain issues